llama_index/readers/psychic.py

```diff
@@ -0,0 +1,73 @@
+"""Psychic reader."""
+import logging
+import os
+from typing import List
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+
+logger = logging.getLogger(__name__)
+
+
+class PsychicReader(BaseReader):
+    """Psychic reader.
+
+    Psychic is a platform that allows syncing data from many SaaS apps through one universal API. 
+    This reader connects to an instance of Psychic and reads data from it, given a connector ID, 
+    account ID, and API key.
+
+    Learn more at docs.psychic.dev.
+
+    Args:
+        psychic_key (str): Secret key for Psychic. Get one at https://dashboard.psychic.dev/api-keys.
+
+    """
+
+    def __init__(self, psychic_key: str = None) -> None:
+        """Initialize with parameters."""
+        try:
+            from psychicapi import Psychic, ConnectorId  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "`psychicapi` package not found, please run `pip install psychicapi`"
+            )
+        if psychic_key is None:
+            psychic_key = os.environ["PSYCHIC_SECRET_KEY"]
+            if psychic_key is None:
+                raise ValueError(
+                    "Must specify `psychic_key` or set environment "
+                    "variable `PSYCHIC_SECRET_KEY`."
+                )
+
+        self.psychic = Psychic(secret_key=psychic_key)
+        self.ConnectorId = ConnectorId
+
+    def load_data(self, connector_id: str = None, account_id: str = None) -> List[Document]:
```

Comment:
To clarify. This reads "all data" from data sources configured in the Psychic dashboard? Curious is there away to configure which subset of data is read?

Reply:
It reads only the data associated with a certain account (a Psychic user's user) and only for a specific source, not all sources.

We also have metadata filters, but it's only partially implemented right now.

Reply:
Cool, thanks for clarifying!